src/Kupo/App.hs

```diff
@@ -417,6 +434,22 @@ consumer tr inputManagement notifyTip mailbox patternsVar Database{..} =
         unstableWindow =
             getLongestRollback longestRollback
 
+
+rollForwardUntil
+    :: forall m block.
+        ( MonadSTM m
+        , MonadLog m
+        , Monad (DBTransaction m)
+        , IsBlock block
+        )
+    => Point
+    -> ApplyFn m block
+rollForwardUntil until tr inputManagement notifyTip database blks patterns = do
+    let blocksBefore = takeWhile ((< until) . getPoint . snd) $ toList blks
+    case nonEmpty blocksBefore of
```

Comment:
That `toList` and `nonEmpty` are awkward, given that `blks` here is already a `NonEmptyList`. Not a big deal but I'll probably tweak that so consider this comment a reminder.